esmvalcore/cmor/_fixes/common.py

```diff
@@ -108,49 +109,70 @@ def fix_metadata(self, cubes):
 class OceanFixGrid(Fix):
     """Fixes for tos, siconc in FGOALS-g3."""
 
-    def fix_data(self, cube):
-        """
-        Fix data.
-
-        Calculate missing latitude/longitude boundaries using interpolation.
-        Based on a similar fix for BCC-CSM2-MR.
+    def fix_metadata(self, cubes):
+        """Fix ``latitude`` and ``longitude`` (metadata and bounds).
 
         Parameters
         ----------
-        cube: iris.cube.Cube
-            Input cube to fix.
+        cubes : iris.cube.CubeList
+            Input cubes.
 
         Returns
         -------
-        iris.cube.Cube
+        iris.cube.CubeList
+
         """
-        rlat = cube.coord('grid_latitude').points
-        rlon = cube.coord('grid_longitude').points
-
-        # Guess coordinate bounds in rlat, rlon (following BCC-CSM2-MR-1).
-        rlat_idx_bnds = np.zeros((len(rlat), 2))
-        rlat_idx_bnds[:, 0] = np.arange(len(rlat)) - 0.5
-        rlat_idx_bnds[:, 1] = np.arange(len(rlat)) + 0.5
-        rlat_idx_bnds[0, 0] = 0.
-        rlat_idx_bnds[len(rlat) - 1, 1] = len(rlat)
-        rlon_idx_bnds = np.zeros((len(rlon), 2))
-        rlon_idx_bnds[:, 0] = np.arange(len(rlon)) - 0.5
-        rlon_idx_bnds[:, 1] = np.arange(len(rlon)) + 0.5
+        cube = self.get_cube_from_list(cubes)
+
+        # Get dimensional coordinates. Note:
+        # - First dimension i -> X-direction (longitude; = dimension 2 in data)
+        # - Second dimension j -> Y-direction (latitude; = dimension 1 in data)
+        i_coord = cube.coord(dim_coords=True, dimensions=2)
+        j_coord = cube.coord(dim_coords=True, dimensions=1)
+
+        # Fix metadata of coordinate i
+        i_coord.var_name = 'i'
+        i_coord.standard_name = None
+        i_coord.long_name = 'cell index along first dimension'
+        i_coord.units = '1'
+        i_coord.circular = False
+
+        # Fix metadata of coordinate j
+        j_coord.var_name = 'j'
+        j_coord.standard_name = None
+        j_coord.long_name = 'cell index along second dimension'
+        j_coord.units = '1'
+
+        # Guess bounds of coordinates i and j if necessary
+        if not i_coord.has_bounds():
+            i_coord.guess_bounds()
+        if not j_coord.has_bounds():
+            j_coord.guess_bounds()
+
+        # Transform i and j bounds into array indices [0, 1, 2, ...]
+        i_to_idx = InterpolatedUnivariateSpline(i_coord.points,
+                                                np.arange(len(i_coord.points)),
+                                                k=1)
+        j_to_idx = InterpolatedUnivariateSpline(j_coord.points,
+                                                np.arange(len(j_coord.points)),
+                                                k=1)
+        i_idx_bnds = i_to_idx(i_coord.bounds)
+        j_idx_bnds = j_to_idx(j_coord.bounds)
```

Comment:
Ah, yes. In this case you're right. I didn't realize that even though the points are not equidistant anymore the bounds are still placed in the center of two points (to be honest I didn't check the values in detail). This of course also translates to the interpolated indices and is exactly what I would expect.

I still think that the interpolation is more general and can also handle cases where the bounds are not located in the center of two points. However, if you think this is too general and this case will never happen than I'm happy to implement your solution :+1:

Reply:
Keep in mind that this fix assumes that `grid_latitude` and `grid_longitude` are completely wrong. If they were reasonable, we wouldn't want to have them overwritten by the conflicting `latitude`, `longitude` information. I think the interpolation can never give better results because the input data is wrong, but it gives the reader the impression that it does something. Imho this is more confusing than helpful. Do you have an example where you would like to apply this fix **and** there is something useful in the original `grid_latitude`, `grid_longitude`?

Reply:
All right, I change the points and bounds of the index coordinates :+1: Any further comments on this? I think I addressed comments.